lib/_tls_wrap.js

```diff
@@ -444,9 +444,7 @@ TLSSocket.prototype.renegotiate = function(options, callback) {
   }
   if (!this._handle.renegotiate()) {
     if (callback) {
-      process.nextTick(function() {
-        callback(new Error('Failed to renegotiate'));
-      });
+      process.nextTick(callback, new Error('Failed to renegotiate'));
```

Comment:
TBH the old error wasn't much better than just throwing a string. At least here you get a pointer to where the error originated.